include/z64item.h

```diff
@@ -253,18 +253,18 @@ typedef enum {
     /* 0x69 */ ITEM_MEDALLION_SPIRIT,
     /* 0x6A */ ITEM_MEDALLION_SHADOW,
     /* 0x6B */ ITEM_MEDALLION_LIGHT,
-    /* 0x6C */ ITEM_KOKIRI_EMERALD,
-    /* 0x6D */ ITEM_GORON_RUBY,
-    /* 0x6E */ ITEM_ZORA_SAPPHIRE,
+    /* 0x6C */ ITEM_SPSTONE_KOKIRI_EMERALD,
+    /* 0x6D */ ITEM_SPSTONE_GORON_RUBY,
+    /* 0x6E */ ITEM_SPSTONE_ZORA_SAPPHIRE,
```

Comment:
or just dont even need to say stone, they are fine as kokiri emerald, goron ruby, and zoras sapphire imo

Reply:
I dont see that as an issue at all, but I also still dont think it needs a common prefix anyway. The names are descriptive on their own

Reply:
using prefixes is an attempt at documenting the "range checks" on item ids
```(item >= ITEM_SPSTONE_KOKIRI_EMERALD) && (item <= ITEM_SPSTONE_ZORA_SAPPHIRE))```
if those were `ITEM_STONE_` then it would make it seem like `ITEM_STONE_OF_AGONY` is part of that "group"

Reply:
Im not sure if prefixing the names in the actual enum is a solution to this problem. Especially if it makes the names less ideal than they need to be

I would think a better solution would be to just put comments in the enum itself to mark off the important groups that someone would need to be mindful of when shifting

Reply:
for example, I would find this more helpful to point out the groups that can be range checked
```
    /* 0x6A */ ITEM_MEDALLION_SHADOW,
    /* 0x6B */ ITEM_MEDALLION_LIGHT,

    // Spirtual Stone Group
    /* 0x6C */ ITEM_KOKIRI_EMERALD,
    /* 0x6D */ ITEM_GORON_RUBY,
    /* 0x6E */ ITEM_ZORA_SAPPHIRE,

    /* 0x6F */ ITEM_STONE_OF_AGONY,
    /* 0x70 */ ITEM_GERUDO_CARD,
```

Reply:
There are situations where you have "sub-groups" as well. Swords are an example, you can have actual swords, but also the "+ Deku Stick + Hammer" group, and there are different functions for each.

Reply:
I think the structure of the ItemId enum is more important than we think and it's worth highlighting it

How would comments look?

Reply:
I agree the structure of it is important, but I think that shoehorning that importance into the names of the enum values itself is probably not the route to take.
I posted an example above, not sure if it didnt load for you or something

Reply:
@EllipticEllipsis that kind of grouping is pretty far removed from the range checking type of grouping we are talking about here. player does treat things like stick and hammer as swords, but I dont think its relevant to their placement/order in the item enum

Reply:
It matters for how they're related to the AP enum, which is what I've been looking at. There are functions that do similar range checks:
```c
s32 func_8008F2BC(Player* this, s32 actionParam) {
    s32 sword;

    sword = PLAYER_SWORD(MASTER);
    if ((actionParam == PLAYER_AP_LAST_USED) ||
        ((sword = actionParam - PLAYER_AP_SWORD_MASTER, (sword >= PLAYER_SWORD(MASTER))) &&
         (sword <= PLAYER_SWORD(BGS)))) {
        return sword;
    }
    return -1;
}```

Reply:
Oh hang on, let me get master's version, that'll be clearer:
```c
s32 func_8008F2BC(Player* this, s32 actionParam) {
    s32 sword = 0;

    if (actionParam != PLAYER_AP_LAST_USED) {
        sword = actionParam - PLAYER_AP_SWORD_MASTER;
        if ((sword < 0) || (sword >= 3)) {
            goto return_neg;
        }
    }

    return sword;

return_neg:
    return -1;
}
```